src/main/java/docSharing/filter/PermissionFilter.java

```diff
@@ -50,166 +56,207 @@ public class PermissionFilter extends GenericFilterBean {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, ExceptionMessage.WRONG_SEARCH.toString());
+        }
 
-            if (request.getParameter(Params.DOCUMENT_ID.toString()) != null) { // checks if we got any parameters
-                Long docId = Long.valueOf(request.getParameter(Params.DOCUMENT_ID.toString()));
-                Long userId = Validations.validateToken(token);
-                UserDocument userDocument = getUserDocument(docId, userId);
+        chain.doFilter(request, response);
+    }
 
-                if (!flag && list.contains("file")) {
-                    if (httpRequest.getMethod().equals(HttpMethod.PATCH.toString())) {// relocate / rename
-                        if (userDocument.getPermission().equals(Permission.MODERATOR)) {
-                            flag = true;
-                        }
-                    }
-                    if (httpRequest.getMethod().equals(HttpMethod.DELETE.toString()) && !(userDocument.getDocument().getUser().getId().equals(userId))) {// only the creator can delete a file
+
+    /**
+     * actionThatDontRequirePermission checks if the given request from the client doesn't need to check for permission,
+     * i.e:  get path of a file doesn't need permission, so it bypass our filter.
+     *
+     * @param list - list of URI request - httpRequest.getRequestURI().split("/").
+     * @return - if the action need to bypass the filter
+     */
+    public boolean actionThatDontRequirePermission(ServletRequest request, List<String> list) {
+        logger.info("in PermissionFilter -> actionThatDontRequirePermission");
+
+        HttpServletRequest httpRequest = (HttpServletRequest) request;
+
+        if (list.contains("sharedDocuments") || list.contains("share") || list.contains("auth") ||
+                list.contains("getAll") || list.contains("getPath") || list.contains("ws") ||
+                list.contains("getUser") || list.contains("getContent") || list.contains("onlineUsers") ||
+                list.contains("import") || list.contains("export") || list.contains("isExists")||
+                httpRequest.getMethod().equals(HttpMethod.OPTIONS.toString()) || httpRequest.getMethod().equals(HttpMethod.GET.toString())) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * filter checks when we get a permission change request from client,
+     * checks if we can grant access to the user's request.
+     *
+     * @return - if the given request can go through the filter.
+     */
+    public boolean actionIsChangePermission(ServletRequest request) throws ResponseStatusException {
+        logger.info("in PermissionFilter -> actionIsChangePermission");
+
+        HttpServletRequest httpRequest = (HttpServletRequest) request;
+        String token = httpRequest.getHeader("authorization");
+        Long docId = Long.valueOf(request.getParameter(Params.DOCUMENT_ID.toString()));
+        Long userId = Validations.validateToken(token);
```

Comment:
We have a dedicated validation service for it. Use authService to validate the token instead of using the direct static method.
It's a bad habit.